[Localization] Localization arena flyout (Active Battle Effects)

EnochG1 opened the pull request:
## What are the changes?
Add the option to localize message text of Arena Flyout UI

## Why am I doing these changes?
All text should be localizable.
And without this task, move names and effect descriptions will be displayed in different languages.

## What did change?
 - Change the code for localize > src/ui/arena-flyout.ts
 - Added locale files, config, localization keys in each language folders.
 - Translated them to Korean.

### Screenshots/Videos
#### En
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/2fc97cca-bfda-4fe2-9685-f190d30ad0a2)

#### Es
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/eabae2b7-6179-4f8b-b249-b7ce645055b1)

#### It
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/3c4c8488-1d24-442b-8465-39a397048804)

#### Fr
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/65f94bba-7885-4da8-99a2-71311ecabc14)

#### De
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/b0ee5a3a-ee37-4499-bd80-6d47ae79a789)
Note : Tested with Harsh Sun due to request.

#### Pt_BR
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/38c846ff-2058-403b-baad-b422c840deba)

#### Zh_CN
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/2545539b-c8dd-4019-b27e-9f7bbbcecc91)

#### Ko
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/9772cd85-aac6-49bc-87b2-e324018fc496)

## How to test the changes?
Manually.
We can override moves related with Arena tags, Weathers, Terrains.
examples)
```Typescript
export const PASSIVE_ABILITY_OVERRIDE: Abilities = Abilities.DROUGHT; // weather ability
export const OPP_PASSIVE_ABILITY_OVERRIDE: Abilities = Abilities.MISTY_SURGE; // terrain ability
export const MOVESET_OVERRIDE: Array<Moves> = [ Moves.RAIN_DANCE, Moves.HAPPY_HOUR, Moves.STEALTH_ROCK, Moves.TOXIC_SPIKES ];
export const OPP_MOVESET_OVERRIDE: Array<Moves> = [ Moves.REFLECT, Moves.TAILWIND, Moves.TRICK_ROOM, Moves.AURORA_VEIL ];
```


## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

CodeTappert commented:
Arent a lot of them already translated in other files? Why Not use them

EnochG1 commented:
> Arent a lot of them already translated in other files? Why Not use them
@CodeTappert,
It's not always same with it (moves.name, terrain, weather).

For example, I wanted to remove this part of original code.
![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/5c8d54b6-7dc5-49da-aa23-092651a70f0b)

40chyan commented on src/locales/zh_CN/arena-flyout.ts:
```diff
@@ -0,0 +1,49 @@
+
+  // WeatherType
+  "sunny": "Sunny",
+  "rain": "Rain",
+  "sandstorm": "Sandstorm",
+  "hail": "Hail",
+  "snow": "Snow",
+  "fog": "Fog",
+  "heavyRain": "Heavy Rain",
+  "harshSun": "Harsh Sun",
+  "strongWinds": "Strong Winds",
+
+  // TerrainType
+  "misty": "Misty Terrain",
+  "electric": "Electric Terrain",
+  "grassy": "Grassy Terrain",
+  "psychic": "Psychic Terrain",
+
+  // ArenaTagType
+  "mudSport": "Mud Sport",
+  "waterSport": "Water Sport",
+  "spikes": "Spikes",
+  "toxicSpikes": "Toxic Spikes",
+  "mist": "Mist",
+  "futureSight": "Future Sight",
+  "doomDesire": "Doom Desire",
+  "wish": "Wish",
+  "stealthRock": "Stealth Rock",
+  "stickyWeb": "Sticky Web",
+  "trickRoom": "Trick Room",
+  "gravity": "Gravity",
+  "reflect": "Reflect",
+  "lightScreen": "Light Screen",
+  "auroraVeil": "Aurora Veil",
+  "quickGuard": "Quick Guard",
+  "wideGuard": "Wide Guard",
+  "matBlock": "Mat Block",
+  "craftyShield": "Crafty Shield",
+  "tailwind": "Tailwind",
+  "happyHour": "Happy Hour",
```
```suggestion
  "activeBattleEffects": "当前战斗效果",
  "player": "我方",
  "neutral": "全局",
  "enemy": "敌方",

  // WeatherType
  "sunny": "大晴天",
  "rain": "下雨",
  "sandstorm": "沙暴",
  "hail": "冰雹",
  "snow": "下雪",
  "fog": "起雾",
  "heavyRain": "大雨",
  "harshSun": "大日照",
  "strongWinds": "乱流",

  // TerrainType
  "misty": "薄雾场地",
  "electric": "电气场地",
  "grassy": "青草场地",
  "psychic": "精神场地",

  // ArenaTagType
  "mudSport": "玩泥巴",
  "waterSport": "玩水",
  "spikes": "撒菱",
  "toxicSpikes": "毒菱",
  "mist": "白雾",
  "futureSight": "预知未来",
  "doomDesire": "破灭之愿",
  "wish": "祈愿",
  "stealthRock": "隐形岩",
  "stickyWeb": "黏黏网",
  "trickRoom": "戏法空间",
  "gravity": "重力",
  "reflect": "反射壁",
  "lightScreen": "光墙",
  "auroraVeil": "极光幕",
  "quickGuard": "快速防守",
  "wideGuard": "广域防守",
  "matBlock": "掀榻榻米",
  "craftyShield": "戏法防守",
  "tailwind": "顺风",
  "happyHour": "快乐时光",
```

40chyan reviewed (commented):
zh_cn

CodeTappert commented on src/locales/de/arena-flyout.ts:
```diff
@@ -0,0 +1,49 @@
+  // WeatherType
+  "sunny": "Sunny",
+  "rain": "Rain",
+  "sandstorm": "Sandstorm",
+  "hail": "Hail",
+  "snow": "Snow",
+  "fog": "Fog",
+  "heavyRain": "Heavy Rain",
+  "harshSun": "Harsh Sun",
+  "strongWinds": "Strong Winds",
+
+  // TerrainType
+  "misty": "Misty Terrain",
+  "electric": "Electric Terrain",
+  "grassy": "Grassy Terrain",
+  "psychic": "Psychic Terrain",
+
+  // ArenaTagType
+  "mudSport": "Mud Sport",
+  "waterSport": "Water Sport",
+  "spikes": "Spikes",
+  "toxicSpikes": "Toxic Spikes",
+  "mist": "Mist",
+  "futureSight": "Future Sight",
+  "doomDesire": "Doom Desire",
+  "wish": "Wish",
+  "stealthRock": "Stealth Rock",
+  "stickyWeb": "Sticky Web",
+  "trickRoom": "Trick Room",
+  "gravity": "Gravity",
+  "reflect": "Reflect",
+  "lightScreen": "Light Screen",
+  "auroraVeil": "Aurora Veil",
+  "quickGuard": "Quick Guard",
+  "wideGuard": "Wide Guard",
+  "matBlock": "Mat Block",
+  "craftyShield": "Crafty Shield",
+  "tailwind": "Tailwind",
+  "happyHour": "Happy Hour",
+};
```
```suggestion
import { SimpleTranslationEntries } from "#app/interfaces/locales";

export const arenaFlyout: SimpleTranslationEntries = {
  // Title
  "activeBattleEffects": "Aktive Kampfeffekte",
  "player": "Spieler",
  "neutral": "Neutral",
  "enemy": "Gegner",

  // WeatherType
  "sunny": "Sonne",
  "rain": "Regen",
  "sandstorm": "Sandsturm",
  "hail": "Hagelsturm",
  "snow": "Schneesturm",
  "fog": "Nebel",
  "heavyRain": "Strömender Regen",
  "harshSun": "Sehr starker Sonnenschein",
  "strongWinds": "Luftströmungen",

  // TerrainType
  "misty": "Nebelfeld",
  "electric": "Elektrofeld",
  "grassy": "Grasfeld",
  "psychic": "Psychofeld",

  // ArenaTagType
  "mudSport": "Lehmsuhler",
  "waterSport": "Nassmacher",
  "spikes": "Stachler",
  "toxicSpikes": "Giftspitzen",
  "mist": "Weißnebel",
  "futureSight": "Seher",
  "doomDesire": "Kismetwunsch",
  "wish": "Wunschtraum",
  "stealthRock": "Tarnsteine",
  "stickyWeb": "Klebenetz",
  "trickRoom": "Bizarroraum",
  "gravity": "Erdanziehung",
  "reflect": "Reflektor",
  "lightScreen": "Lichtschild",
  "auroraVeil": "Auroraschleier",
  "quickGuard": "Rapidschutz",
  "wideGuard": "Rundumschutz",
  "matBlock": "Tatami-Schild",
  "craftyShield": "Trickschutz",
  "tailwind": "Rückenwind",
  "happyHour": "Goldene Zeiten",
};
```

Adri1 commented on src/locales/fr/arena-flyout.ts:
```diff
@@ -0,0 +1,49 @@
+
+  // WeatherType
+  "sunny": "Sunny",
+  "rain": "Rain",
+  "sandstorm": "Sandstorm",
+  "hail": "Hail",
+  "snow": "Snow",
+  "fog": "Fog",
+  "heavyRain": "Heavy Rain",
+  "harshSun": "Harsh Sun",
+  "strongWinds": "Strong Winds",
+
+  // TerrainType
+  "misty": "Misty Terrain",
+  "electric": "Electric Terrain",
+  "grassy": "Grassy Terrain",
+  "psychic": "Psychic Terrain",
+
+  // ArenaTagType
+  "mudSport": "Mud Sport",
+  "waterSport": "Water Sport",
+  "spikes": "Spikes",
+  "toxicSpikes": "Toxic Spikes",
+  "mist": "Mist",
+  "futureSight": "Future Sight",
+  "doomDesire": "Doom Desire",
+  "wish": "Wish",
+  "stealthRock": "Stealth Rock",
+  "stickyWeb": "Sticky Web",
+  "trickRoom": "Trick Room",
+  "gravity": "Gravity",
+  "reflect": "Reflect",
+  "lightScreen": "Light Screen",
+  "auroraVeil": "Aurora Veil",
+  "quickGuard": "Quick Guard",
+  "wideGuard": "Wide Guard",
+  "matBlock": "Mat Block",
+  "craftyShield": "Crafty Shield",
+  "tailwind": "Tailwind",
+  "happyHour": "Happy Hour",
```
```suggestion
  "activeBattleEffects": "Effets Actifs",
  "player": "Vous",
  "neutral": "Neutre",
  "enemy": "Ennemi",

  // WeatherType
  "sunny": "Soleil",
  "rain": "Pluie",
  "sandstorm": "Tempête de Sable",
  "hail": "Grêle",
  "snow": "Neige",
  "fog": "Brouillard",
  "heavyRain": "Pluie Battante",
  "harshSun": "Soleil Brulant",
  "strongWinds": "Vent Mystérieux",

  // TerrainType
  "misty": "Champ Brumeux",
  "electric": "Champ Électrifié",
  "grassy": "Champ Herbu",
  "psychic": "Champ Psychique",

  // ArenaTagType
  "mudSport": "Lance-Boue",
  "waterSport": "Tourniquet",
  "spikes": "Picots",
  "toxicSpikes": "Pics Toxik",
  "mist": "Brume",
  "futureSight": "Prescience",
  "doomDesire": "Vœu Destructeur",
  "wish": "Vœu",
  "stealthRock": "Piège de Roc",
  "stickyWeb": "Toile Gluante",
  "trickRoom": "Distorsion",
  "gravity": "Gravité",
  "reflect": "Protection",
  "lightScreen": "Mur Lumière",
  "auroraVeil": "Voile Aurore",
  "quickGuard": "Prévention",
  "wideGuard": "Garde Large",
  "matBlock": "Tatamigaeshi",
  "craftyShield": "Vigilance",
  "tailwind": "Vent Arrière",
  "happyHour": "Étrennes",
```

Adri1 reviewed (commented):
Added French

josericardo-fo commented on src/locales/pt_BR/arena-flyout.ts:
```diff
@@ -0,0 +1,49 @@
+
+export const arenaFlyout: SimpleTranslationEntries = {
+  // Title
+  "activeBattleEffects": "Active Battle Effects",
+  "player": "Player",
+  "neutral": "Neutral",
+  "enemy": "Enemy",
+
+  // WeatherType
+  "sunny": "Sunny",
+  "rain": "Rain",
+  "sandstorm": "Sandstorm",
+  "hail": "Hail",
+  "snow": "Snow",
+  "fog": "Fog",
+  "heavyRain": "Heavy Rain",
+  "harshSun": "Harsh Sun",
+  "strongWinds": "Strong Winds",
+
+  // TerrainType
+  "misty": "Misty Terrain",
+  "electric": "Electric Terrain",
+  "grassy": "Grassy Terrain",
+  "psychic": "Psychic Terrain",
```
```suggestion
  "activeBattleEffects": "Efeitos de Batalha Ativos",
  "player": "Jogador",
  "neutral": "Neutro",
  "enemy": "Inimigo",

  // WeatherType
  "sunny": "Ensolarado",
  "rain": "Chuva",
  "sandstorm": "Tempestade de Areia",
  "hail": "Granizo",
  "snow": "Neve",
  "fog": "Névoa",
  "heavyRain": "Chuva Forte",
  "harshSun": "Sol Forte",
  "strongWinds": "Ventos Fortes",

  // TerrainType
  "misty": "Terreno Enevoado",
  "electric": "Terreno Elétrico",
  "grassy": "Terreno de Planta",
  "psychic": "Terreno Psíquico",
```

returntoice reviewed (approved):
Good for Korean locales

NicusPulcis commented:
yo @EnochG1 i've added the italian translation : )

flx-sta commented on src/utils.ts:
```diff
@@ -409,6 +409,13 @@ export function formatText(unformattedText: string): string {
   return text.join(" ");
 }
 
+export function toLowerCamelString(unformattedText: string): string {
```
`Camel Case` is by definition starting with lower case.
`thisIsCamelCase`

Otherwise it would be `Pascal case`
`ThisIsPascalCase`

torranx replied:
agree, i'd add that String could be removed as well hence something like `toCamelCase`

EnochG1 replied:
Modified it :)

Arxxer commented on src/locales/es/arena-flyout.ts:
```diff
@@ -0,0 +1,49 @@
+  // WeatherType
+  "sunny": "Sunny",
+  "rain": "Rain",
+  "sandstorm": "Sandstorm",
+  "hail": "Hail",
+  "snow": "Snow",
+  "fog": "Fog",
+  "heavyRain": "Heavy Rain",
+  "harshSun": "Harsh Sun",
+  "strongWinds": "Strong Winds",
+
+  // TerrainType
+  "misty": "Misty Terrain",
+  "electric": "Electric Terrain",
+  "grassy": "Grassy Terrain",
+  "psychic": "Psychic Terrain",
+
+  // ArenaTagType
+  "mudSport": "Mud Sport",
+  "waterSport": "Water Sport",
+  "spikes": "Spikes",
+  "toxicSpikes": "Toxic Spikes",
+  "mist": "Mist",
+  "futureSight": "Future Sight",
+  "doomDesire": "Doom Desire",
+  "wish": "Wish",
+  "stealthRock": "Stealth Rock",
+  "stickyWeb": "Sticky Web",
+  "trickRoom": "Trick Room",
+  "gravity": "Gravity",
+  "reflect": "Reflect",
+  "lightScreen": "Light Screen",
+  "auroraVeil": "Aurora Veil",
+  "quickGuard": "Quick Guard",
+  "wideGuard": "Wide Guard",
+  "matBlock": "Mat Block",
+  "craftyShield": "Crafty Shield",
+  "tailwind": "Tailwind",
+  "happyHour": "Happy Hour",
+};
```
```suggestion
import { SimpleTranslationEntries } from "#app/interfaces/locales";

export const arenaFlyout: SimpleTranslationEntries = {
  // Title
  "activeBattleEffects": "Efectos de Terreno Activos",
  "player": "Jugador",
  "neutral": "Neutral",
  "enemy": "Enemigo",

  // WeatherType
  "sunny": "Sol",
  "rain": "Lluvia",
  "sandstorm": "Tormenta de Arena",
  "hail": "Granizo",
  "snow": "Nieve",
  "fog": "Niebla",
  "heavyRain": "Diluvio",
  "harshSun": "Sol Abrasador",
  "strongWinds": "Turbulencias",

  // TerrainType
  "misty": "Campo de Niebla",
  "electric": "Campo Eléctrico",
  "grassy": "Campo de Hierba",
  "psychic": "Campo Psíquico",

  // ArenaTagType
  "mudSport": "Chapoteo Lodo",
  "waterSport": "Hidrochorro",
  "spikes": "Púas",
  "toxicSpikes": "Púas Tóxicas",
  "mist": "Neblina",
  "futureSight": "Premonición",
  "doomDesire": "Deseo Oculto",
  "wish": "Deseo",
  "stealthRock": "Trampa Rocas",
  "stickyWeb": "Red Vidcosa",
  "trickRoom": "Espacio Raro",
  "gravity": "Gravedad",
  "reflect": "Reflejo",
  "lightScreen": "Pantalla de Luz",
  "auroraVeil": "Velo Aurora",
  "quickGuard": "Anticipo",
  "wideGuard": "Vasta Guardia",
  "matBlock": "Escudo Tatami",
  "craftyShield": "Truco Defensa",
  "tailwind": "Viento Afín",
  "happyHour": "Paga Extra",
};
```

Arxxer reviewed (commented):
es-es localization added

EnochG1 commented:
@Arxxer, @CodeTappert, @40chyan, @josericardo-fo, @Adri1, @NicusPulcis 
I added screenshots.
If all is well, please approve.
(You can also request specific tags if you want to see the result.)

@flx-sta, @torranx I renamed it `toCamelCaseString` :)

NicusPulcis commented:
@EnochG1 italian is perfect

NicusPulcis reviewed (commented):
italian is ok
